checks/evaluation/branch_protection.go

```diff
@@ -60,41 +70,117 @@ const (
 )
 
 // BranchProtection runs Branch-Protection check.
-func BranchProtection(name string, dl checker.DetailLogger,
-	r *checker.BranchProtectionsData,
+func BranchProtection(name string,
+	findings []finding.Finding, dl checker.DetailLogger,
 ) checker.CheckResult {
-	var scores []levelScore
+	expectedProbes := []string{
+		blocksDeleteOnBranches.Probe,
+		blocksForcePushOnBranches.Probe,
+		branchProtectionAppliesToAdmins.Probe,
+		dismissesStaleReviews.Probe,
+		requiresApproversForPullRequests.Probe,
+		requiresCodeOwnersReview.Probe,
+		requiresLastPushApproval.Probe,
+		requiresUpToDateBranches.Probe,
+		runsStatusChecksBeforeMerging.Probe,
+		requiresPRsToChangeCode.Probe,
+	}
+
+	if !finding.UniqueProbesEqual(findings, expectedProbes) {
+		e := sce.WithMessage(sce.ErrScorecardInternal, "invalid probe results")
+		return checker.CreateRuntimeErrorResult(name, e)
+	}
+
+	branchScores := make(map[string]*levelScore)
+	warnedBranches := make([]string, 0)
 
-	// Check protections on all the branches.
-	for i := range r.Branches {
-		var score levelScore
-		b := r.Branches[i]
+	for i := range findings {
+		f := &findings[i]
+		if f.Outcome == finding.OutcomeNotApplicable {
+			return checker.CreateInconclusiveResult(name, "unable to detect any development/release branches")
+		}
 
+		var branchName string
+		for k := range f.Values {
+			if k == "branchProtected" || k == "numberOfRequiredReviewers" || k == "codeownersFiles" {
+				continue
+			}
+			branchName = k
+		}
 		// Protected field only indates that the branch matches
 		// one `Branch protection rules`. All settings may be disabled,
 		// so it does not provide any guarantees.
-		protected := !(b.Protected != nil && !*b.Protected)
-		if !protected {
+		protected := (f.Values["branchProtected"] == 1)
```

Comment:
The `protected` value comes from the probe findings. `1` means `true`, `0` means `false`.

Reply:
Added in https://github.com/ossf/scorecard/pull/3759/commits/7d8ce6fd83f2445ed20b08a589eaac21d07d4487